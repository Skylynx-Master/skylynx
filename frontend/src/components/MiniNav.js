import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

const Ul = styled.ul`
  list-style: none;
  display: none;
  flex-flow: row nowrap;

  li {
    padding: 18px 10px;
  }

  @media (max-width: 768px) {
    display: flex;
    flex-flow: column nowrap;
    background-color: #fdcb9e;
    position: fixed;
    transform: ${({ open }) => (open ? "translateX(0)" : "translateX(100%)")};
    top: 0;
    right: 0;
    height: 100vh;
    width: 45%;
    padding-top: 3.5rem;
    transition: transform 0.3s ease-in-out;

    li {
      color: #3f3f44;
      text-align: center;
    }
  }
`;

const MiniNav = ({ open }) => {
  return (
    <Ul open={open}>
      <Link to="/filters">
        <li>Buscar</li>
      </Link>
      <Link to="/dashboard">
        <li>Favoritos</li>
      </Link>
      <Link to="/dashboard">
        <li>Crear Publicaci√≥n</li>
      </Link>
      <Link to="/">
        <li>Inicio</li>
      </Link>
    </Ul>
  );
};

export default MiniNav;
