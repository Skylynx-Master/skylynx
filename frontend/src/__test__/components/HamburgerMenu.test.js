import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import HamburgerMenu from "../../components/HamburgerMenu";
import MiniNav from "../../components/MiniNav";

describe("Header Component Testing", () => {
  const hamburger = mount(
    <BrowserRouter>
      <HamburgerMenu />
    </BrowserRouter>
  );

  test("Hamburger Render", () => {
    expect(hamburger.length).toEqual(1);
  });

  test("Hamburger Lines Render", () => {
    expect(hamburger.find(".IconBurger").length).toEqual(2);
  });

  const mininav = mount(
    <BrowserRouter>
      <MiniNav />
    </BrowserRouter>
  );

  test("MiniNav Menu Availability", () => {
    expect(mininav.length).toEqual(1);
  });
});

describe("HamburgerMenu Snapshot", () => {
  test("Test UI Hamburger Menu component", () => {
    const snaphamburger = create(
      <BrowserRouter>
        <HamburgerMenu />
      </BrowserRouter>
    );
    expect(snaphamburger.toJSON()).toMatchSnapshot();
  });
});
