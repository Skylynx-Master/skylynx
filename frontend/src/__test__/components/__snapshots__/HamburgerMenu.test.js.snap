// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu Snapshot Test UI Hamburger Menu component 1`] = `
Array [
  <div
    className="sc-AxirZ gfqYih IconBurger"
    onClick={[Function]}
    open={false}
  >
    <div />
    <div />
    <div />
  </div>,
  <ul
    className="sc-AxjAm buYwmk"
    open={false}
  >
    <a
      href="/filters"
      onClick={[Function]}
    >
      <li>
        Buscar
      </li>
    </a>
    <a
      href="/dashboard"
      onClick={[Function]}
    >
      <li>
        Favoritos
      </li>
    </a>
    <a
      href="/dashboard"
      onClick={[Function]}
    >
      <li>
        Crear Publicaci√≥n
      </li>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <li>
        Inicio
      </li>
    </a>
  </ul>,
]
`;
